packages/azure-kusto-ingest/src/ingestClientBase.ts

```diff
@@ -31,22 +34,64 @@ export abstract class KustoIngestClientBase extends AbstractKustoClient {
         blob: string | BlobDescriptor,
         ingestionProperties?: IngestionPropertiesInput,
         maxRetries: number = KustoIngestClientBase.MaxNumberOfRetryAttempts
-    ): Promise<QueueSendMessageResponse> {
+    ): Promise<IngestionResult> {
         this.ensureOpen();
 
         const props = this._getMergedProps(ingestionProperties);
 
         const descriptor = blob instanceof BlobDescriptor ? blob : new BlobDescriptor(blob);
+
+        const authorizationContext = await this.resourceManager.getAuthorizationContext();
+        const ingestionBlobInfo = new IngestionBlobInfo(descriptor, props, authorizationContext);
+
+        const reportToTable = props.reportLevel !== ReportLevel.DoNotReport && props.reportMethod !== ReportMethod.Queue;
+
+        const time = Date.now().toString();
+        if (reportToTable) {
+            const statusTableUri = await this.resourceManager.getStatusTable();
+            const statusTableClient = createStatusTableClient(statusTableUri![0].uri);
+
+            const status = {
```

Comment:
This should probably be a method since you repeat a lot with the next object creation